{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#INPUT FILES\n",
    "\n",
    "The VWF NNK experiment was split into three different runs and the generated high throughput sequencing files have been deposited to the NCBI Sequence Read Archive (project accession number #PRJNA356764). #PRJNA296481 encompasses 3 sets of 50bp paired-end .fastq files used in the following pipelines:\n",
    "\n",
    "- Set 1: #SRR5097080\n",
    "- Set 2: #SRR5097081\n",
    "- Set 3: #SRR5097082"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Pipeline for set 1\n",
    "\n",
    "###*clean_fastq_ORF.py*\n",
    "    - seed match to position the read\n",
    "    - sense-antisense read match at NNK\n",
    "    - minimum read quality of 5 on the scale of 0-41"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/usr/bin/env python\n",
    "# k2rt 05-11-2013\n",
    "# \n",
    "#import possible modules the script might need\n",
    "import sys\n",
    "import re\n",
    "import mod575\n",
    "\n",
    "# read in the file numb\n",
    "number = sys.argv[1]\n",
    "#open both fastq files\n",
    "fileName1=sys.argv[2]\n",
    "fileName2=sys.argv[3]\n",
    "file1=open(fileName1,'r')\n",
    "file2=open(fileName2,'r')\n",
    "#open output file for peptides\n",
    "fileName5=str(number)+\"_sense\"\n",
    "fileName6=str(number)+\"_antisense\"\n",
    "outfile3=open(fileName5,'w')\n",
    "outfile4=open(fileName6,'w')\n",
    "#create a little stats file over all the reads to know when we are throwing reads out\n",
    "stats=open('stats_clean_fastq.txt','a')\n",
    "#some counters to keep track of the running file and collect numbers for stats\n",
    "counter=0\n",
    "good_reads=0\n",
    "unmatched_pairs=0\n",
    "seed1=0\n",
    "seed2=0\n",
    "seed3=0\n",
    "seed4=0\n",
    "seed_unmatched=0\n",
    "bad_quality=0\n",
    "sense_antisense_unmatched=0\n",
    "OOF=0\n",
    "\n",
    "#define a generator to return 4 lines from each paired read file\n",
    "def get_read_set(file1, file2):\n",
    "\t'''Expects to open files as input. Returns 8 lines of data, four from\n",
    "each file. Return characters are stripped from each line.\n",
    "'''\n",
    "\twhile True:\n",
    "\t\tline1 = file1.readline().strip()\n",
    "\t\tline2 = file1.readline().strip()\n",
    "\t\tline3 = file1.readline().strip()\n",
    "\t\tline4 = file1.readline().strip()\n",
    "\t\tline5 = file2.readline().strip()\n",
    "\t\tline6 = file2.readline().strip()\n",
    "\t\tline7 = file2.readline().strip()\n",
    "\t\tline8 = file2.readline().strip()\n",
    "\t\tif (line1 and line2 and line3 and line4 and line5 and line6 and line7 and line8):\n",
    "\t\t\tyield line1, line2, line3, line4, line5, line6, line7, line8\n",
    "\t\telse:\n",
    "\t\t\tbreak\n",
    "\n",
    "#start going through the fastq files 8 lines at a time using the generator\n",
    "for pairedRead in get_read_set(file1, file2):\n",
    "\tcounter+=1\n",
    "\trow1=pairedRead[0]\n",
    "\trow2=pairedRead[1]\n",
    "\trow3=pairedRead[2]\n",
    "\trow4=pairedRead[3]\n",
    "\trow5=pairedRead[4]\n",
    "\trow6=pairedRead[5]\n",
    "\trow7=pairedRead[6]\n",
    "\trow8=pairedRead[7]\n",
    "#check if the @header matches between paired fastqfiles, if not count the error and continue\n",
    "\thead1=row1.split()\n",
    "\thead2=row5.split()\n",
    "\tif not head1[0]==head2[0]:\n",
    "\t\tunmatched_pairs+=1\n",
    "\t\tcontinue\n",
    "#check both pairs of reads for matching of at least one of three seeds of 8bp. If no match, goes to junk\n",
    "\n",
    "\tif re.match(r'^[ATCGN]{12}GAGCAGG[ATCGN]+',row2):\n",
    "\t\tsense=row2[27:45]\n",
    "\t\tantisense=row6[29:47]\n",
    "\t\tsense_quality=row4[27:45]\n",
    "\t\tantisense_quality=row8[29:49]\n",
    "\t\tsense_row=row2\n",
    "\t\tantisense_row=row6\n",
    "\t\tseed1+=1\n",
    "\telif re.match(r'^[ATCGN]{19}CGCCCAAC[ATCGN]+',row2):\n",
    "\t\tsense=row2[27:45]\n",
    "\t\tantisense=row6[29:47]\n",
    "\t\tsense_quality=row4[27:45]\n",
    "\t\tantisense_quality=row8[29:49]\n",
    "\t\tsense_row=row2\n",
    "\t\tantisense_row=row6\n",
    "\t\tseed2+=1\n",
    "\telif re.match(r'^[ATCGN]{11}ATCAGAGGC[ATCGN]+',row2):\n",
    "\t\tsense=row6[27:45]\n",
    "\t\tantisense=row2[29:47]\n",
    "\t\tsense_quality=row8[27:45]\n",
    "\t\tantisense_quality=row4[29:49]\n",
    "\t\tsense_row=row6\n",
    "\t\tantisense_row=row2\n",
    "\t\tseed3+=1\n",
    "\telif re.match(r'^[ATCGN]{20}AGGATTTCC[ATCGN]+',row2):\n",
    "\t\tsense=row6[27:45]\n",
    "\t\tantisense=row2[29:47]\n",
    "\t\tsense_quality=row8[27:45]\n",
    "\t\tantisense_quality=row4[29:49]\n",
    "\t\tsense_row=row6\n",
    "\t\tantisense_row=row2\n",
    "\t\tseed4+=1\n",
    "\telse:\n",
    "\t\tif re.match(r'^[ATCGN]{12}GAGCAGG[ATCGN]+',row6):\n",
    "\t\t\tsense=row6[27:45]\n",
    "\t\t\tantisense=row2[29:47]\n",
    "\t\t\tsense_quality=row8[27:45]\n",
    "\t\t\tantisense_quality=row4[29:49]\n",
    "\t\t\tsense_row=row6\n",
    "\t\t\tantisense_row=row2\n",
    "\t\t\tseed1+=1\n",
    "\t\telif re.match(r'^[ATCGN]{19}CGCCCAAC[ATCGN]+',row6):\n",
    "\t\t\tsense=row6[27:45]\n",
    "\t\t\tantisense=row2[29:47]\n",
    "\t\t\tsense_quality=row8[27:45]\n",
    "\t\t\tantisense_quality=row4[29:49]\n",
    "\t\t\tsense_row=row6\n",
    "\t\t\tantisense_row=row2\n",
    "\t\t\tseed2+=1\n",
    "\t\telif re.match(r'^[ATCGN]{11}ATCAGAGGC[ATCGN]+',row6):\n",
    "\t\t\tsense=row2[27:45]\n",
    "\t\t\tantisense=row6[29:47]\n",
    "\t\t\tsense_quality=row4[27:45]\n",
    "\t\t\tantisense_quality=row8[29:49]\n",
    "\t\t\tsense_row=row2\n",
    "\t\t\tantisense_row=row6\n",
    "\t\t\tseed3+=1\n",
    "\t\telif re.match(r'^[ATCGN]{20}AGGATTTCC[ATCGN]+',row6):\n",
    "\t\t\tsense=row2[27:45]\n",
    "\t\t\tantisense=row6[29:47]\n",
    "\t\t\tsense_quality=row4[27:45]\n",
    "\t\t\tantisense_quality=row8[29:49]\n",
    "\t\t\tsense_row=row2\n",
    "\t\t\tantisense_row=row6\n",
    "\t\t\tseed4+=1\t\n",
    "\t\telse:\n",
    "\t\t\tseed_unmatched+=1\n",
    "\t\t\tcontinue\n",
    "\n",
    "\t#checking for matching\n",
    "\tif not sense==mod575.reverse_compliment(antisense):\n",
    "\t\tsense_antisense_unmatched+=1\n",
    "\t\tcontinue\n",
    "\n",
    "\n",
    "#checking for quality. translating the quality code from ASCII33 and removing any read where the pos quality is lower than 5 on scale from 0-41!\n",
    "\tstatus=0\n",
    "\tfor pos in sense_quality:\n",
    "\t\tscore=int(ord(str(pos)))-33\n",
    "\t\tif score < 5:\n",
    "\t\t\tstatus+=1\n",
    "\tfor pos in antisense_quality:\n",
    "\t\tscore=int(ord(str(pos)))-33\n",
    "\t\tif score < 5:\n",
    "\t\t\tstatus+=1\n",
    "\tif status !=0:\n",
    "\t\tbad_quality+=1\n",
    "\t\tcontinue\t\t\n",
    "\n",
    "\tgood_reads+=1\n",
    "\tprint >> outfile3, sense_row\n",
    "\tprint >> outfile4, antisense_row\n",
    "\tcontinue\t\t\n",
    "\n",
    "\n",
    "print >> stats,number,counter,good_reads,unmatched_pairs,seed1,seed2,seed3,seed4,seed_unmatched,sense_antisense_unmatched,bad_quality\n",
    "\n",
    "file1.close()\n",
    "file2.close()\n",
    "outfile3.close()\n",
    "outfile4.close()\n",
    "stats.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "STATS (72976223/84083495 => ~86.8%):\n",
    "\n",
    "NNK 84083495 72976223 0 78209684 246784 941149 5519 4680359 6113876 313037"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###*barcode.py*:\n",
    "    - reads in the filtered sense/antisense reads\n",
    "    - seperates the reads by 6 different barcodes\n",
    "\n",
    "In set 1 , barcodes 1 and 2 are controls, 3 and 4 are ADAMTS13 treated libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/usr/bin/env python\n",
    "# k2rt 05-11-2013\n",
    "# \n",
    "#import possible modules the script might need\n",
    "import sys\n",
    "import re\n",
    "import mod575\n",
    "\n",
    "#open both fastq files\n",
    "fileName1=sys.argv[1]\n",
    "fileName2=sys.argv[2]\n",
    "file1=open(fileName1,'r')\n",
    "file2=open(fileName2,'r')\n",
    "#open output file for peptides\n",
    "outfile1=open('1_sense','w')\n",
    "outfile2=open('2_sense','w')\n",
    "outfile3=open('3_sense','w')\n",
    "outfile4=open('4_sense','w')\n",
    "outfile5=open('5_sense','w')\n",
    "outfile6=open('6_sense','w')\n",
    "\n",
    "#create a little stats file over all the reads to know when we are throwing reads out\n",
    "stats=open('stats_barcode.txt','a')\n",
    "#some counters to keep track of the running file and collect numbers for stats\n",
    "counter=0\n",
    "good_reads=0\n",
    "seed1=0\n",
    "seed2=0\n",
    "seed3=0\n",
    "seed4=0\n",
    "seed5=0\n",
    "seed6=0\n",
    "\n",
    "seed_unmatched=0\n",
    "\n",
    "\n",
    "#define a generator to return 4 lines from each paired read file\n",
    "def get_read_set(file1, file2):\n",
    "\t'''Expects to open files as input. Returns 1 line of data, 1 from\n",
    "each file. Return characters are stripped from each line.\n",
    "'''\n",
    "\twhile True:\n",
    "\t\tline1 = file1.readline().strip()\n",
    "\t\tline2 = file2.readline().strip()\n",
    "\n",
    "\t\tif (line1 and line2):\n",
    "\t\t\tyield line1, line2\n",
    "\t\telse:\n",
    "\t\t\tbreak\n",
    "\n",
    "#start going through the fastq files 8 lines at a time using the generator\n",
    "for pairedRead in get_read_set(file1, file2):\n",
    "\tcounter+=1\n",
    "\trow1=pairedRead[0]\n",
    "\trow2=pairedRead[1]\n",
    "\n",
    "#check both pairs of reads for matching of at least one of three seeds of 8bp. If no match, goes to junk\n",
    "\n",
    "\tif re.match(r'^[ATCGN]{7}GAATC[ATCGN]+',row1):\n",
    "\t\tseed1+=1\n",
    "\t\tprint >> outfile1, row1\n",
    "\t\t\n",
    "\telif re.match(r'^[ATCGN]{6}GTCTA[ATCGN]+',row2):\n",
    "\t\tseed1+=1\n",
    "\t\tprint >> outfile1, row1\n",
    "\t\t\n",
    "\telif re.match(r'^[ATCGN]{7}TCTGA[ATCGN]+',row1):\n",
    "\t\tseed2+=1\n",
    "\t\tprint >> outfile2, row1\n",
    "\t\t\n",
    "\telif re.match(r'^[ATCGN]{6}CAGAG[ATCGN]+',row2):\n",
    "\t\tseed2+=1\n",
    "\t\tprint >> outfile2, row1\n",
    "\t\t\n",
    "\telif re.match(r'^[ATCGN]{7}CAGTC[ATCGN]+',row1):\n",
    "\t\tseed5+=1\n",
    "\t\tprint >> outfile5, row1\n",
    "\t\t\n",
    "\telif re.match(r'^[ATCGN]{6}TTCAC[ATCGN]+',row2):\n",
    "\t\tseed5+=1\n",
    "\t\tprint >> outfile5, row1\n",
    "\t\t\n",
    "\telif re.match(r'^[ATCGN]{7}AGTCT[ATCGN]+',row1):\n",
    "\t\tseed6+=1\n",
    "\t\tprint >> outfile6, row1\n",
    "\t\n",
    "\telif re.match(r'^[ATCGN]{6}AGTCT[ATCGN]+',row2):\n",
    "\t\tseed6+=1\n",
    "\t\tprint >> outfile6, row1\n",
    "\t\n",
    "\telif re.match(r'^[ATCGN]{7}TGCAA[ATCGN]+',row1):\n",
    "\t\tseed3+=1\n",
    "\t\tprint >> outfile3, row1\n",
    "\t\n",
    "\telif re.match(r'^[ATCGN]{6}TCAGG[ATCGN]+',row2):\n",
    "\t\tseed3+=1\n",
    "\t\tprint >> outfile3, row1\n",
    "\t\t\n",
    "\telif re.match(r'^[ATCGN]{7}ACAGT[ATCGN]+',row1):\n",
    "\t\tseed4+=1\n",
    "\t\tprint >> outfile4, row1\n",
    "\t\n",
    "\telif re.match(r'^[ATCGN]{6}ATGTC[ATCGN]+',row2):\n",
    "\t\tseed4+=1\n",
    "\t\tprint >> outfile4, row1\n",
    "\t\n",
    "\telse:\n",
    "\t\tseed_unmatched+=1\n",
    "\n",
    "print >> stats,counter,seed1,seed2,seed3,seed4,seed5,seed6,seed_unmatched\n",
    "\n",
    "file1.close()\n",
    "file2.close()\n",
    "outfile1.close()\n",
    "outfile2.close()\n",
    "outfile3.close()\n",
    "outfile4.close()\n",
    "outfile5.close()\n",
    "outfile6.close()\n",
    "stats.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "STATS:\n",
    "\n",
    "all barcode1 barcode2 barcode3 barcode4 barcode5 barcode6 nomatch\n",
    "72976223 18164303 18103794 18723277 17976287 2374 3526 2662\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###*count_peptides.py*:\n",
    "    - counts the unique peptides\n",
    "    - counts the unique DNA sequencing encoding that peptide\n",
    "    - sorts peptides by count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/usr/bin/env python\n",
    "# k2rt 08-10-2012\n",
    "# this script recognizes specifically the sense strand of Colin's data\n",
    "\n",
    "# import possible modules the script might need\n",
    "import sys\n",
    "import re\n",
    "import mod575\n",
    "from collections import defaultdict\n",
    "\n",
    "# read in the file numb\n",
    "number = sys.argv[1]\n",
    "fileName1=number+'_sense'\n",
    "fileName2=number+'_peptide'\n",
    "input=open(fileName1,'r')\n",
    "outfile=open(fileName2,'w')\n",
    "\n",
    "peptides=dict()\n",
    "sequences=defaultdict(defaultdict)\n",
    "counter=0\n",
    "\n",
    "for line in input:\n",
    "\tcounter+=1\n",
    "\tline=line.strip()\n",
    "\tsense=line[27:45]\n",
    "\tpeptide=mod575.translate_dna(sense)\n",
    "\tif peptide in peptides:\n",
    "\t\tpeptides[peptide]+=1\n",
    "\telse:\n",
    "\t\tpeptides[peptide]=1\n",
    "\n",
    "\tif peptide in sequences:\n",
    "\t\tif sense in sequences[peptide]:\n",
    "\t\t\tsequences[peptide][sense]+=1\n",
    "\t\telse:\n",
    "\t\t\tsequences[peptide][sense]=1\n",
    "\telse:\n",
    "\t\tsequences[peptide][sense]=1\n",
    "\n",
    "for peptide in sorted(peptides, key=peptides.get, reverse=True):\n",
    "\tseqlist=[]\t\n",
    "\tfor sense in sequences[peptide]:\n",
    "\t\tseqdata=str(sense)+':'+str(sequences[peptide][sense])\n",
    "\t\tseqlist.append(seqdata)\n",
    "\tseqs=';'.join(seqlist)\n",
    "\tdata=peptide+'\\t'+str(peptides[peptide])+'\\t'+seqs\n",
    "\tprint >> outfile, data\n",
    "\n",
    "input.close()\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#grep command to remove peptides containing stop codon\n",
    "grep -v 'X' 1_peptide > 1_noX_peptide\n",
    "etc."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "|Treatment|Seq_name|Reads|Percent|Pep_name|Peptide_count|Peptide_noX|\n",
    "|-|-|-|-|-|-|-|\n",
    "|Control1_set1|1_sense|18,164,303|24.9%|1_peptide|529,198|486,090|\n",
    "|Control2_set1|2_sense|18,103,794|24.8%|2_peptide|518,022|476,329|\n",
    "|ATS1_set1|3_sense|18,723,277|25.7%|3_peptide|570,009|521,164|\n",
    "|ATS2_set1|4_sense|17,976,287|24.6%|4_peptide|620,543|563,065|\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Pipeline for set 2\n",
    "\n",
    "###*clean_fastq_ORF.py*:\n",
    "    - Used the same script as above\n",
    "\n",
    "STATS:\n",
    "\n",
    "Lane1 95835969 58240300 0 61496916 310103 717407 9280 33302263 4074659 218747\n",
    "\n",
    "###*barcode.py*:\n",
    "    - Used the same script as above\n",
    "\n",
    "STATS:\n",
    "all **barcode1 barcode2** barcode3 barcode4 **barcode5** barcode6 nomatch\n",
    "58240300 **17795244 20224913** 74473 120151 **20020332** 4291 896\n",
    "\n",
    "In set 2 , barcodes 1 and 2 are controls and 5 is the ADAMTS13 treated library\n",
    "\n",
    "###*count_peptides.py*:\n",
    "    - Used the same script as above"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "|Treatment|Seq_name|Reads|Percent|Pep_name|Peptide_count|Peptide_noX|\n",
    "|-|-|-|-|-|-|-|\n",
    "|Control1_set2|1_sense|17795244|30.6%|1_peptide|500,808|459,596|\n",
    "|Control2_set2|2_sense|20224913|34.7%|2_peptide|520,348|476,524|\n",
    "|ATS1_set2|5_sense|20020332|34.4%|5_peptide|697,037|624,528|\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Pipeline for set 3\n",
    "\n",
    "###*clean_fastq_ORF.py*:\n",
    "    - Used the same script as above\n",
    "\n",
    "STATS:\n",
    "\n",
    "Lane2 90262326 54627695 0 57861693 200158 614395 7386 31578694 3889895 166042\n",
    "\n",
    "\n",
    "###*barcode.py*:\n",
    "    - Used the same script as above\n",
    "\n",
    "STATS:\n",
    "all barcode1 barcode2** barcode3 barcode4 **barcode5 barcode6 nomatch\n",
    "54627695 1736868 2242992 **19193241 23267645** 6189067 1992756 5126\n",
    "\n",
    "In set 3 , barcodes 3 and 4 are ADAMTS13 treated libraries\n",
    "\n",
    "###*count_peptides.py*:\n",
    "    - Used the same script as above"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "|Treatment|Seq_name|Reads|Percent|Pep_name|Peptide_count|Peptide_noX|\n",
    "|-|-|-|-|-|-|-|\n",
    "|ATS1_set3|3_sense|19193241|35.1%|3_peptide|562,951|512,064|\n",
    "|ATS1_set3|4_sense|23267645|42.6%|4_peptide|673,966|606,714|\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Merge all above peptide counts for DESEQ2\n",
    "\n",
    "*merge_9.py*:\n",
    "    - merges petides with at least 4 counts over all the samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/usr/bin/env python\n",
    "# k2rt 08-10-2012\n",
    "# this script recognizes specifically the sense strand of Colin's data\n",
    "\n",
    "# import possible modules the script might need\n",
    "import sys\n",
    "import re\n",
    "\n",
    "# read in the file numb and create the input output files\n",
    "treatment = sys.argv[1]\n",
    "fileName1=sys.argv[2]\n",
    "fileName2=sys.argv[3]\n",
    "fileName3=sys.argv[4]\n",
    "fileName4=sys.argv[5]\n",
    "fileName5=sys.argv[6]\n",
    "fileName6=sys.argv[7]\n",
    "fileName7=sys.argv[8]\n",
    "fileName8=sys.argv[9]\n",
    "fileName9=sys.argv[10]\n",
    "\n",
    "sumlimit=int(sys.argv[11])\n",
    "input1=open(fileName1,'r')\n",
    "input2=open(fileName2,'r')\n",
    "input3=open(fileName3,'r')\n",
    "input4=open(fileName4,'r')\n",
    "input5=open(fileName5,'r')\n",
    "input6=open(fileName6,'r')\n",
    "input7=open(fileName7,'r')\n",
    "input8=open(fileName8,'r')\n",
    "input9=open(fileName9,'r')\n",
    "\n",
    "outfile=open('DESeq/'+treatment+'_merged','w')\n",
    "#make up necessary dictionaries\n",
    "peptides=dict()\n",
    "pep1=dict()\n",
    "pep2=dict()\n",
    "pep3=dict()\n",
    "pep4=dict()\n",
    "pep5=dict()\n",
    "pep6=dict()\n",
    "pep7=dict()\n",
    "pep8=dict()\n",
    "pep9=dict()\n",
    "\n",
    "#read in the peptides in the two files\n",
    "\n",
    "print >> outfile, 'peptide\\tuntreated1\\tuntreated2\\tuntreated3\\tuntreated4\\ttreated1\\ttreated2\\ttreated3\\ttreated4\\ttreated5'\n",
    "\n",
    "for line in input1:\n",
    "\tline=line.strip()\n",
    "\trow=line.split()\n",
    "\tif row[0] in peptides:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpeptides[row[0]]=1\n",
    "\tif row[0] in pep1:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpep1[row[0]]=row[1]\n",
    "for line in input2:\n",
    "\tline=line.strip()\n",
    "\trow=line.split()\n",
    "\tif row[0] in peptides:\n",
    "\t\tpeptides[row[0]]+=1\n",
    "\telse:\n",
    "\t\tpeptides[row[0]]=1\n",
    "\tif row[0] in pep2:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpep2[row[0]]=row[1]\n",
    "\n",
    "for line in input3:\n",
    "\tline=line.strip()\n",
    "\trow=line.split()\n",
    "\tif row[0] in peptides:\n",
    "\t\tpeptides[row[0]]+=1\n",
    "\telse:\n",
    "\t\tpeptides[row[0]]=1\n",
    "\tif row[0] in pep3:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpep3[row[0]]=row[1]\n",
    "for line in input4:\n",
    "\tline=line.strip()\n",
    "\trow=line.split()\n",
    "\tif row[0] in peptides:\n",
    "\t\tpeptides[row[0]]+=1\n",
    "\telse:\n",
    "\t\tpeptides[row[0]]=1\n",
    "\tif row[0] in pep4:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpep4[row[0]]=row[1]\n",
    "for line in input5:\n",
    "\tline=line.strip()\n",
    "\trow=line.split()\n",
    "\tif row[0] in peptides:\n",
    "\t\tpeptides[row[0]]+=1\n",
    "\telse:\n",
    "\t\tpeptides[row[0]]=1\n",
    "\tif row[0] in pep5:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpep5[row[0]]=row[1]\n",
    "\n",
    "for line in input6:\n",
    "\tline=line.strip()\n",
    "\trow=line.split()\n",
    "\tif row[0] in peptides:\n",
    "\t\tpeptides[row[0]]+=1\n",
    "\telse:\n",
    "\t\tpeptides[row[0]]=1\n",
    "\tif row[0] in pep6:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpep6[row[0]]=row[1]\n",
    "\n",
    "for line in input7:\n",
    "\tline=line.strip()\n",
    "\trow=line.split()\n",
    "\tif row[0] in peptides:\n",
    "\t\tpeptides[row[0]]+=1\n",
    "\telse:\n",
    "\t\tpeptides[row[0]]=1\n",
    "\tif row[0] in pep7:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpep7[row[0]]=row[1]\n",
    "\n",
    "for line in input8:\n",
    "\tline=line.strip()\n",
    "\trow=line.split()\n",
    "\tif row[0] in peptides:\n",
    "\t\tpeptides[row[0]]+=1\n",
    "\telse:\n",
    "\t\tpeptides[row[0]]=1\n",
    "\tif row[0] in pep8:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpep8[row[0]]=row[1]\n",
    "\n",
    "for line in input9:\n",
    "\tline=line.strip()\n",
    "\trow=line.split()\n",
    "\tif row[0] in peptides:\n",
    "\t\tpeptides[row[0]]+=1\n",
    "\telse:\n",
    "\t\tpeptides[row[0]]=1\n",
    "\tif row[0] in pep9:\n",
    "\t\tprint 'sth weird'\n",
    "\telse:\n",
    "\t\tpep9[row[0]]=row[1]\n",
    "\n",
    "for peptide in peptides:\n",
    "\tsum=0\n",
    "\tout=list()\n",
    "\tout.append(peptide)\n",
    "\tif peptide in pep1:\n",
    "\t\tout.append(pep1[peptide])\n",
    "\t\tcount_pep1=int(pep1[peptide])\n",
    "\telse:\n",
    "\t\tout.append('0')\n",
    "\t\tcount_pep1=0\n",
    "\tif peptide in pep2:\n",
    "\t\tout.append(pep2[peptide])\n",
    "\t\tcount_pep2=int(pep2[peptide])\n",
    "\telse:\n",
    "\t\tout.append('0')\n",
    "\t\tcount_pep2=0\n",
    "\tif peptide in pep3:\n",
    "\t\tout.append(pep3[peptide])\n",
    "\t\tcount_pep3=int(pep3[peptide])\n",
    "\telse:\n",
    "\t\tout.append('0')\n",
    "\t\tcount_pep3=0\n",
    "\tif peptide in pep4:\n",
    "\t\tout.append(pep4[peptide])\n",
    "\t\tcount_pep4=int(pep4[peptide])\n",
    "\telse:\n",
    "\t\tout.append('0')\n",
    "\t\tcount_pep4=0\n",
    "\tif peptide in pep5:\n",
    "\t\tout.append(pep5[peptide])\n",
    "\t\tcount_pep5=int(pep5[peptide])\n",
    "\telse:\n",
    "\t\tout.append('0')\n",
    "\t\tcount_pep5=0\n",
    "\n",
    "\tif peptide in pep6:\n",
    "\t\tout.append(pep6[peptide])\n",
    "\t\tcount_pep6=int(pep6[peptide])\n",
    "\telse:\n",
    "\t\tout.append('0')\n",
    "\t\tcount_pep6=0\n",
    "\n",
    "\tif peptide in pep7:\n",
    "\t\tout.append(pep7[peptide])\n",
    "\t\tcount_pep7=int(pep7[peptide])\n",
    "\telse:\n",
    "\t\tout.append('0')\n",
    "\t\tcount_pep7=0\n",
    "\n",
    "\tif peptide in pep8:\n",
    "\t\tout.append(pep8[peptide])\n",
    "\t\tcount_pep8=int(pep8[peptide])\n",
    "\telse:\n",
    "\t\tout.append('0')\n",
    "\t\tcount_pep8=0\n",
    "\n",
    "\tif peptide in pep9:\n",
    "\t\tout.append(pep9[peptide])\n",
    "\t\tcount_pep9=int(pep9[peptide])\n",
    "\telse:\n",
    "\t\tout.append('0')\n",
    "\t\tcount_pep9=0\n",
    "\n",
    "\tout='\\t'.join(out)\n",
    "\tsum=count_pep1+count_pep2+count_pep3+count_pep4+count_pep5+count_pep6+count_pep7+count_pep8+count_pep9\n",
    "\n",
    "\tif sum<sumlimit:\n",
    "\t\tcontinue\n",
    "\telse:\n",
    "\t\tprint >> outfile, out\n",
    "\n",
    "input1.close()\n",
    "input2.close()\n",
    "input3.close()\n",
    "input4.close()\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###DESEQ2\n",
    "\n",
    "The merged file is used as an input for R based package DESeq2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#DESEQ2\n",
    "\n",
    "library(DESeq2)\n",
    "data=read.table('ALL_VWF73_4_merged',header=TRUE,row.names=1)\n",
    "countData=as.matrix(data)\n",
    "colData=data.frame(condition=factor(c(\"control\",\"control\",\"control\",\"control\",\"treatement\",\"treatement\",\"treatement\",\"treatement\",\"treatement\")))\n",
    "dds=DESeqDataSetFromMatrix(countData, colData, formula(~ condition))\n",
    "\n",
    "dds=DESeq(dds)\n",
    "res=results(dds)\n",
    "\n",
    "pdf('ALL_VWF73_4_deseq2_plotMA.pdf')\n",
    "plotMA(dds,alpha=.05,ylim=c(-2,2),main='ALL_ADA_4')\n",
    "dev.off()\n",
    "\n",
    " write.table(res,file=\"ALL_VWF73_4_deseq2.txt\", col.names=FALSE, append=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All reads with adjusted p-value <=0.05 is considered significantly enriched/depleted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#to extract significant peptides from the DESEQ2 output file\n",
    "awk '$3>0' ALL_VWF73_4_deseq2.txt | awk '$7<=0.05' | sort -g -k 7 > enriched_top.txt\n",
    "awk '$3<0' ALL_VWF73_4_deseq2.txt | awk '$7<=0.05' | sort -g -k 7 > depleted_top.txt\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
